_pytest/python_api.py

```diff
@@ -584,10 +584,11 @@ def raises(expected_exception, *args, **kwargs):
 
     """
     __tracebackhide__ = True
-    for exc in filterfalse(isclass, always_iterable(expected_exception)):
-        msg = ("exceptions must be old-style classes or"
-               " derived from BaseException, not %s")
-        raise TypeError(msg % type(exc))
+    if not isclass(expected_exception) or not issubclass(expected_exception, BaseException):
+        for exc in filterfalse(isclass, always_iterable(expected_exception)):
```

Comment:
please try using `base_type=type`

Reply:
oh, that sounds nice! just a moment.

Reply:
just to be clear: do you want ``base_type=type`` or ``base_type=(type, six.text_type, six.binary_type)``?

Reply:
it works in both variants. :)

upon further inspection, i'd prefer the longer variant as i believe it to lead to nicer error messages as strings don't get split into individual characters.

```python
base_type = (type, text_type, binary_type)
for exc in filterfalse(isclass, always_iterable(expected_exception, base_type)):
    msg = ("exceptions must be old-style classes or"
           " derived from BaseException, not %s")
    raise TypeError(msg % type(exc))
```

however, please feel free to modify the PR to your liking. ^^

Reply:
and (sorry for the many messages): thank you for reviewing and commenting! it's greatly appreciated. :D

Reply:
you are doing great work :+1: